sdk/src/main/java/com/mapbox/maps/Snapshotter.kt

```diff
@@ -25,29 +23,24 @@ import kotlin.math.min
 /**
  * [Snapshotter] is high-level component responsible for taking map snapshot with given [MapSnapshotOptions].
  */
-open class Snapshotter {
+open class Snapshotter(
+  context: Context,
+  options: MapSnapshotOptions,
+  private val snapshotOverlayOptions: SnapshotOverlayOptions = SnapshotOverlayOptions()
+) {
 
-  private val context: WeakReference<Context>
-  private val coreSnapshotter: MapSnapshotterInterface
-  private val pixelRatio: Float
-  private val mapSnapshotOptions: MapSnapshotOptions
-  private val snapshotOverlayOptions: SnapshotOverlayOptions
+  private val context: WeakReference<Context> = WeakReference(context)
+  private var coreSnapshotter: MapSnapshotterInterface?
```

Comment:
can we make it a val?
```suggestion
  private val coreSnapshotter: MapSnapshotterInterface = MapSnapshotter(options)
```

Reply:
how can we then null it in `destroy`? 😄 
and actually make this fix work

Reply:
just realised that we are setting it to null during cancel. 

btw I noticed there's a `finalize` method on MapSnapshotter class, is it something we can utilise to clean it up?

Reply:
checked it - same leak 😆 😆 😆

Reply:
Got it, LGTM. 
One suggestion, instead of using `!!`, shall we use ?.let and throw some meaningful exception, like IllegalStateException ("snapshotter has been destroyed") etc?

Reply:
And please add tests for it as well🙂